python/ray/dag/tests/experimental/test_accelerated_dag.py

```diff
@@ -2493,6 +2493,91 @@ async def main():
     compiled_dag.teardown()
 
 
+class TestVisualization:
+    def test_visualize_basic(self, ray_start_regular):
+        @ray.remote
+        class Actor:
+            def echo(self, x):
+                return x
+
+        actor = Actor.remote()
+
+        with InputNode() as i:
+            dag = actor.echo.bind(i)
+
+        compiled_dag = dag.experimental_compile()
+
+        # Call the visualize method
+        filename = "compiled_graph_basic"
+        compiled_dag.visualize(filename=filename, view=False)
+
+        assert os.path.exists(f"{filename}.png"), "Visualization file was not created."
+        res = compiled_dag.execute(1)
+        ray.get(res)
+        compiled_dag.teardown()
+
+        os.remove(f"{filename}.png")
+        os.remove(f"{filename}")
+
+    def test_visualize_multi_return(self, ray_start_regular):
+        @ray.remote
+        class Actor:
+            @ray.method(num_returns=2)
+            def return_two(self, x):
+                return x, x + 1
+
+        actor = Actor.remote()
+
+        with InputNode() as i:
+            o1, o2 = actor.return_two.bind(i)
+            dag = MultiOutputNode([o1, o2])
+
+        compiled_dag = dag.experimental_compile()
+
+        filename = "compiled_graph_multi_return"
+        compiled_dag.visualize(filename=filename, view=False)
+
+        assert os.path.exists(f"{filename}.png"), "Visualization file was not created."
+        res = compiled_dag.execute(1)
+        ray.get(res)
+        # Clean up
+        compiled_dag.teardown()
+
+        os.remove(f"{filename}.png")
+        os.remove(f"{filename}")
+
+    def test_visualize_multi_return2(self, ray_start_regular):
+        @ray.remote
+        class Actor:
+            @ray.method(num_returns=2)
+            def return_two(self, x):
+                return x, x + 1
+
+            def echo(self, x):
+                return x
+
+        a = Actor.remote()
+        b = Actor.remote()
+        with InputNode() as i:
+            o1, o2 = a.return_two.bind(i)
+            o3 = b.echo.bind(o1)
+            o4 = b.echo.bind(o2)
+            dag = MultiOutputNode([o3, o4])
+
+        compiled_dag = dag.experimental_compile()
+
+        filename = "compiled_graph_multi_return2"
+        compiled_dag.visualize(filename=filename, view=False)
+
+        assert os.path.exists(f"{filename}.png"), "Visualization file was not created."
```

Comment:
We should not be checking the existence of visualization files (which does not give much signal), but the graph structures (nodes and edges)

Reply:
changed to check nodes and edges with doc